### DISCUSSION TITLE
`import vtk` takes long to analyze, speed up/cache?

### DISCUSSION THREAD

**Discussion by nschloe** [microsoft/pyright#3164] [Q&A]
When opening a file that contains nothing more than
```python
import vtk
```
my vim config fires up pyright that tries to analyze the file. This takes a very long time, presumably because VTK isn't typed and pyright tries to infer all types in this large library.

Is there a way to cache the results, or perhaps turn it off for certain libs?

**Comment by erictraut** [comment]
Pyright lazily evaluates the types of symbols that you import from a library, so a simple `import vtk` statement should be very cheap. It will do almost no work until you reference a symbol from the library, such as `x = vtk.foo`, and then it does the minimal work necessary to evaluate the type of `foo`.

Furthermore, by default pyright doesn't attempt to infer any types from an untyped library, so even if you directly reference a symbol from `vtk`, it will be evaluated as `Unknown` at no cost. Perhaps you have enabled the `useLibraryCodeForTypes` option? If so, you could disable that.

I tried to repro the issue you're seeing in VS code, and I don't see any problem with `import vtk` even if `useLibraryCodeForTypes` is enabled. The code in `vtk.py` is very simple.

**Reply by nschloe** [reply]
Thanks Eric for looking into this. Setting `useLibraryCodeForTypes = false` didn't do anything for me, but `diagnosticMode = 'openFilesOnly'` did.
